### ISSUE TITLE
Oxygen Generator stopped working?

### ISSUE THREAD

**Issue by Tsunder** [Regalis11/Barotrauma#8974]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Vanilla multiplayer campaign, just started on Herja,

About the 2nd round in people started suffocating (no o2 in room). Oxygen generator was not refilling o2 tanks either (appeared stuck at 0%). o2 vents appeared to be running (they were spinning)

Grid load appeared normal, everyone had to wear masks/suits.

We tried multiple things including disconnecting and reconnecting the power wire.

Some time later after a moloch attack and sinking and repairing the sub, I went back to the o2 generator -- it had full o2 tanks now and the rooms had air.

**Steps To Reproduce**
If possible, describe how the developers can reproduce the issue. It is often extremely hard to fix a bug if we don't know how to make it happen.

**Version**
0.17.12.0 stable, multiplayer

**Additional information**
potentially related to #8344 

host was alive at the time of o2 problem discovery iirc, but died due to barotrauma during the repairs/sinking, along with 1 more member of the 6 man crew we had.

routing condition out of o2 gen to a text display showed it had 100 condition

**Comment by Regalis11** [comment]
Could also be caused by https://github.com/regalis11/barotrauma/issues/8873 if someone did any modifications to the wiring

**Comment by Tsunder** [comment]
fresh campaign, don't think any power related wiring was done yet.

a light component or two from unrelated systems may have been wrenched off prior.


the machine should have gotten flooded/take water damage during the moloch attach, but I'm not sure how much or if it needed repairs after. I did record the session and will watch the recording <24h

**Comment by Regalis11** [comment]
Most likely related to https://github.com/Regalis11/Barotrauma/issues/9269 - the hull oxygen desync would definitely be explained by it, but it shouldn't have affected the O2 generator unless someone did modifications to the wiring

**Comment by NilanthAnimosus** [comment]
Tested against bugfixes commit https://github.com/Regalis11/Barotrauma-development/commit/6efb667a91d39ee7715a094a89a32ed20b73d6ec

I tried numerous things to try to desync wiring, first off I couldn't get two clients to interact with the same wiring, and rewiring both ends had no effect. I tried to see if I could desync a fabricator alongside an oxygen generator.

I mid-round rewired things through relays and that didn't do anything unexpected to cause a desync issue, powering stuff through the backup power batteries on another submarine. 

I also tried wrenching off some random light components and in-use relays from backup power arrays on the humpback that were being utilized - and I still had no power issues (Which I tested with crafting and oxygen tank recharging), be it all the batteries connected through one or three relays.

and disconnecting/reconnecting a device seems fine too (as mentioned in an older ticket #8873 (Both same version) which I am inclined by the dates of the tickets being reported that this old issue was likely caused by a similar issue that may have been fixed by https://github.com/Regalis11/Barotrauma-development/commit/f762dbb24bc98d5c964eccb9a6e08e8edabb8270
